### PR TITLE
Add Key Concepts page

### LINKED ISSUE #816: How to configure COSMOS to talk to external hardware
**Is your feature request related to a problem? Please describe.**
I'm trying to setup COSMOS to talk to an external piece of hardware (which does not reside in docker). I've followed the [Getting Started](https://docs.openc3.com/docs/v5/gettingstarted) section, creating my own plugin with my target definition and my own interface definition, specifying the proper UDP ports for my hardware, however COSMOS never receives any telemetry from my hardware.

The demo defines an interface which communicates with `host.docker.internal` on ports 8080/8081. Since my target is outside of docker, it seems likely that I need to provide a port mapping from the host into the docker environment somewhere, but I'm not sure where that should go. Maybe in [`compose.yml`](https://github.com/OpenC3/cosmos-project/blob/d3b62aea354762100e3bd189ac7a0a5828096723/compose.yaml)??, although even then its not 100% clear which service it should be applied to.

**Describe the solution you'd like**
I think talking to hardware outside of docker is likely the more common use case for docker, so it would be helpful if the tutorial walked through that case, rather than the case where you're talking to a target within docker (which seems like it'd be a niche use case). 

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by jmthomas**
I'll need to regenerate the site before merge but check this out and let me know what you think. This is the result of a few recent discussions we've had about confusion about the architecture and things we're doing.

closes #816

### CODE REVIEW

**Comment by jmthomas** [comment]
closes #961

File: docs.openc3.com/docs/getting-started/key_concepts.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+---
+sidebar_position: 4
+title: Key Concepts
+---
+
+# OpenC3 COSMOS Key Concepts
+
+## Projects
+
+The main COSMOS [repo](https://github.com/OpenC3/cosmos) contains all the source code used to build and run COSMOS. However, users (not developers) of COSMOS should use the COSMOS [project](https://github.com/OpenC3/cosmos-project) to launch COSMOS. The project consists of the [openc3.sh](https://github.com/OpenC3/cosmos-project/blob/main/openc3.sh) and [openc3.bat](https://github.com/OpenC3/cosmos-project/blob/main/openc3.bat) files for starting and stopping COSMOS, the [compose.yaml](https://github.com/OpenC3/cosmos-project/blob/main/compose.yaml) for configuring the COSMOS containers, and the [.env](https://github.com/OpenC3/cosmos-project/blob/main/.env) file for setting runtime variables.
```
Comment (ryanmelt) [inline docs.openc3.com/docs/getting-started/key_concepts.md:10]: Additionally it contains user modifiable config files for both Redis and Traefik.

File: docs.openc3.com/docs/getting-started/key_concepts.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+| [openc3-ruby](https://github.com/OpenC3/cosmos/blob/main/openc3-ruby/Dockerfile)                                         | Alpine image containing Ruby and Python                                   |
+| [openc3-base](https://github.com/OpenC3/cosmos/blob/main/openc3/Dockerfile)                                              | Builds on openc3-ruby and includes all the COSMOS source code             |
+| [openc3-node](https://github.com/OpenC3/cosmos/blob/main/openc3-node/Dockerfile)                                         | Builds on openc3-base and includes NodeJS for COSMOS frontend development |
+| [openc3-cosmos-init](https://github.com/OpenC3/cosmos/blob/main/openc3-cosmos-init/Dockerfile)                           | Builds on openc3-node, copies all the COSMOS tools and builds them        |
+| [openc3-operator](https://github.com/OpenC3/cosmos/blob/main/openc3-operator/Dockerfile)                                 | Sets up the base image to run microservice_operator.rb                    |
```
Comment (ryanmelt) [inline docs.openc3.com/docs/getting-started/key_concepts.md:26]: Inconsistent - does not mention builds on openc3-base

File: docs.openc3.com/docs/getting-started/key_concepts.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+| cosmos-openc3-traefik-1                  | Provides a reverse proxy and load balancer with routes to the COSMOS endpoints                         |
+
+The image list for [Enterprise COSMOS](https://openc3.com/enterprise) consists of the following:
+
+| Name                                  | Description                                              |
+| ------------------------------------- | -------------------------------------------------------- |
+| cosmos-enterprise-openc3-metrics-1    | Rails server that provides metrics on COSMOS performance |
+| cosmos-enterprise-openc3-keycloak-1   | Single-Sign On service for authentication                |
+| cosmos-enterprise-openc3-postgresql-1 | SQL Database for use by Keycloak                         |
+
```
Comment (ryanmelt) [inline docs.openc3.com/docs/getting-started/key_concepts.md:67]: Also includes nfs in Kubernetes.

File: docs.openc3.com/docs/getting-started/key_concepts.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+| cosmos-openc3-operator-1                 | Main COSMOS container that runs the interfaces and target mircoservices                                |
+| cosmos-openc3-cosmos-cmd-tlm-api-1       | Rails server that provides all the COSMOS API endpoints                                                |
+| cosmos-openc3-cosmos-script-runner-api-1 | Rails server that provides the Script API endpoints                                                    |
+| cosmos-openc3-redis-1                    | Serves the static target configuration                                                                 |
+| cosmos-openc3-redis-ephemeral-1          | Serves the [streams](https://redis.io/docs/data-types/streams) containing the raw and decomutated data |
+| cosmos-openc3-minio-1                    | Provides a S3 like bucket storage interface and also serves as a static webserver for the tool files   |
+| cosmos-openc3-traefik-1                  | Provides a reverse proxy and load balancer with routes to the COSMOS endpoints                         |
+
+The image list for [Enterprise COSMOS](https://openc3.com/enterprise) consists of the following:
```
Comment (ryanmelt) [inline docs.openc3.com/docs/getting-started/key_concepts.md:60]: containers.

**Review by ryanmelt — CHANGES_REQUESTED** [review]
Overall good new content!
I don't really like differentiating images and containers.  I think that just confuses users.  I would just describe each of the containers, and not differentiate.

**Pull request merged**